src/stores/AppStore.js

```diff
@@ -433,19 +433,30 @@ export default types
      *
      * @param {*} taskObject
      */
-    function assignTask(taskObject) {
+    function assignTask(taskObject, isPrevious) {
       if (taskObject && !Utils.Checkers.isString(taskObject.data)) {
         taskObject = {
           ...taskObject,
           data: JSON.stringify(taskObject.data),
         };
       }
+
       self.task = Task.create(taskObject);
       if (self.taskHistory.findIndex((x) => x.taskId === self.task.id) === -1) {
-        self.taskHistory.push({
-          taskId: self.task.id,
-          annotationId: null,
-        });
+        if (isPrevious) {
+          self.taskHistory = [
+            {
+              taskId: self.task.id,
+              annotationId: null,
+            },
+            ...self.taskHistory,
+          ];
```

Comment:
Is `taskHistory` a reactive object in mobx, or just a standard array? I would use the builtin array properties if its a mobx reactive value as I believe those are overridden in the case of mobx for reactive bindings propagation and replacing the array could mess up reactivity.

```suggestion
          self.taskHistory.unshift({
            taskId: self.task.id,
            annotationId: null,
          })
```